internal/kzg/domain.go

```diff
@@ -1,197 +0,0 @@
-	}
-
-	return domain
-}
-
-// BitReverse applies the bit-reversal permutation to `list`.
-// `len(list)` must be a power of 2
-//
-// This is in no way needed for basic KZG and is included in this library as
-// a stepping-stone to full Dank-sharding.
-//
-//
-//
-/*
-Taken from a chat with Dr Dankrad:
-- Samples are going to be contiguous when we switch on full sharding.
-- Technically there is nothing that requires samples to be contiguous
-pieces of data, but it seems a lot nicer.
-- also the relationship between original and interpolated data would
-look really strange, with them being interleaved.
-- Everything is just nice in brp and looks really strange in direct order
-once you introduce sharding. So best to use it from the start and not have
-to think about all these when you add DAS.
-*/
-//
-// Modified from [gnark-crypto](https://github.com/ConsenSys/gnark-crypto/blob/8f7ca09273c24ed9465043566906cbecf5dcee91/ecc/bls12-381/fr/fft/fft.go#L245)
-//
-// [bit_reverse](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#reverse_bits)
-func bitReverse[K interface{}](list []K) {
-	n := uint64(len(list))
-	if !utils.IsPowerOfTwo(n) {
-		panic("size of list must be a power of two")
-	}
-
-	nn := uint64(64 - bits.TrailingZeros64(n))
-
-	for i := uint64(0); i < n; i++ {
-		irev := bits.Reverse64(i) >> nn
-		if irev > i {
-			list[i], list[irev] = list[irev], list[i]
-		}
-	}
-}
-
-// Bit reverses the elements in the domain
-// and their inverses
-//
-// [bit_reversal_permutation](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#bit_reversal_permutation)
-func (domain *Domain) ReverseRoots() {
-	bitReverse(domain.Roots)
-	bitReverse(domain.PreComputedInverses)
-}
-
-// Returns the index of the element in the domain or -1 if it
-// is not an element in the domain
-func (domain Domain) findRootIndex(point fr.Element) int {
-	for i := 0; i < int(domain.Cardinality); i++ {
-		if point.Equal(&domain.Roots[i]) {
-			return i
-		}
-	}
-
-	return -1
-}
-
-// Evaluates a lagrange polynomial and returns an error if the
-// number of evaluations in the polynomial is different to the size
-// of the domain
-//
-// [evaluate_polynomial_in_evaluation_form](https://github.com/ethereum/consensus-specs/blob/3a2304981a3b820a22b518fe4859f4bba0ebc83b/specs/deneb/polynomial-commitments.md#evaluate_polynomial_in_evaluation_form)
-func (domain *Domain) EvaluateLagrangePolynomial(poly Polynomial, evalPoint fr.Element) (*fr.Element, error) {
-	outputPoint, _, err := domain.evaluateLagrangePolynomial(poly, evalPoint)
-
-	return outputPoint, err
-}
-
-// Evaluates polynomial and returns the index of the evaluation point
-// in the domain, if it is a point in the domain and -1 otherwise
-// This semantics was copied from the go library, see: https://cs.opensource.google/go/x/exp/+/522b1b58:slices/slices.go;l=117
```

Comment:
Perhaps this comment belongs in `findRootIndex()`? Or perhaps it's fine here too.